addons/pos_restaurant/static/src/overrides/components/navbar/navbar.js

```diff
@@ -72,21 +72,64 @@ patch(Navbar.prototype, {
         this.pos.floorPlanStyle = mode;
     },
     newFloatingOrder() {
-        this.pos.add_new_order();
+        const order = this.pos.add_new_order();
+        order.setBooked(true);
         this.pos.showScreen("ProductScreen");
     },
     getFloatingOrders() {
-        return this.pos.get_open_orders().filter((order) => !order.table_id);
+        return this.pos
+            .get_open_orders()
+            .filter((order) => !order.table_id)
+            .sort((a, b) => {
+                const noteA = a.note || "";
+                const noteB = b.note || "";
+                if (noteA && noteB) {
+                    // Both have notes
+                    const timePattern = /^\d{1,2}:\d{2}/;
+
+                    const aHasTime = timePattern.test(noteA);
+                    const bHasTime = timePattern.test(noteB);
+
+                    if (aHasTime && bHasTime) {
+                        // Both have times, compare by time
+                        const [aTime] = noteA.match(timePattern);
+                        const [bTime] = noteB.match(timePattern);
+                        const [aHour, aMinute] = aTime.split(":");
+                        const [bHour, bMinute] = bTime.split(":");
+                        const formattedATime = aHour.padStart(2, "0") + aMinute.padStart(2, "0");
+                        const formattedBTime = bHour.padStart(2, "0") + bMinute.padStart(2, "0");
+                        return formattedATime.localeCompare(formattedBTime);
+                    } else if (aHasTime) {
+                        // a has time, b does not
+                        return -1;
+                    } else if (bHasTime) {
+                        // b has time, a does not
+                        return 1;
+                    } else {
+                        // Neither have times, compare alphabetically
+                        return noteA.localeCompare(noteB);
+                    }
```

Comment:
for this to work it means that users would have manually added a time in the order name ?

Reply:
Yep the user needs to start its name by hh:mm. I'm talking with mobt to see if we could add a time picker in master but for 17.4 the plan is to keep it like this

Reply:
ok, i see. The code is fine, but it seems a bit complicated. Couldn't we just sort alphabetically, given that the time is at the beggining  ?
```suggestion
                    const timePattern = /^\d{1,2}:\d{2}/;

                    const [aTime] = noteA.match(timePattern);
                    const [bTime] = noteB.match(timePattern);
                    if (aTime && !bTime || bTime && !aTime) {
                        return aHasTime ? -1: 1;
                    } 
                    return noteA.localeCompare(noteB);
                    
```